File: backend/src/main/java/be/technobel/parsemaster/service/declaration/AuthService.java
package be.technobel.parsemaster.service.declaration;

import be.technobel.parsemaster.dto.AuthDTO;
import be.technobel.parsemaster.form.ChangePasswordForm;
import be.technobel.parsemaster.form.LoginForm;
import be.technobel.parsemaster.form.RequestPasswordForm;
import jakarta.servlet.http.HttpServletRequest;

public interface AuthService {
  AuthDTO signIn(LoginForm form);

  void signOut(HttpServletRequest request);

  void requestPassword(RequestPasswordForm form);

  void changePassword(ChangePasswordForm form, String token);

  boolean existsPasswordToken(String token);
}

File: src/main/kotlin/be/technobel/playzone/bll/services/AuthService.kt
package be.technobel.playzone.bll.services

import be.technobel.playzone.bll.exceptions.NotFoundException
import be.technobel.playzone.pl.models.forms.LoginForm
import jakarta.servlet.http.HttpServletRequest
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.core.AuthenticationException

interface AuthService {
	@Throws(AuthenticationException::class)
	fun login(form: LoginForm): String

	@Throws(BadCredentialsException::class)
	fun logout(request: HttpServletRequest)

	/** Envoie un mail de récupération de mot de passe */
	fun passwordModificationRequest(email: String)

	/** Modifie le mot de passe de l'utilisateur pour le token en paramètre */
	@Throws(NotFoundException::class)
	fun passwordModification(token: String, password: String)
}